### ISSUE TITLE
False positive for rule `strict-effect-handlers`

### ISSUE THREAD

**Issue by 7iomka** [effector/eslint-plugin#149]
![image](https://github.com/effector/eslint-plugin/assets/18473137/62c990bf-1727-4b51-b2a3-743f0ebd90f8)

<img width="445" alt="image" src="https://github.com/effector/eslint-plugin/assets/18473137/22f231c5-2b48-4b8f-b557-e69c9a04a656">
<img width="393" alt="image" src="https://github.com/effector/eslint-plugin/assets/18473137/203fe4a3-c7fa-4c10-8d22-f0680a252d7e">

`Function `Unknown` can lead to scope loosing in Fork API.eslint[effector/strict-effect-handlers](https://eslint.effector.dev/rules/strict-effect-handlers)`

And this is not just IDE bug
![image](https://github.com/effector/eslint-plugin/assets/18473137/483a244f-4caf-48b5-9161-d37932de3b20)

**Comment by 7iomka** [comment]
Maybe this is related with updates of plugin or eslint
Versions on the left side don't have the issue.

![image](https://github.com/effector/eslint-plugin/assets/18473137/de7a6c33-5937-4027-a863-64ccc20fc574)

**Comment by igorkamyshev** [comment]
Hello! Thanks for report.

Could you please ensure that TS integration for ESLint is set up correctly? (refer to the issue #131)

I believe in this case ESLint fallen back to JS-checks which could be inaccurate and could not check that the field of object (`$$editAdress.setAsDefaultFx`) is Effect.

**Comment by igorkamyshev** [comment]
It cannot be reproduced — https://stackblitz.com/edit/effector-vite-react-template-dajnx8?file=tsconfig.json,src%2Fmodel.ts%3AL20,src%2Fanother-model.ts,src%2Fapplication.tsx

**Comment by 7iomka** [comment]
> It cannot be reproduced — https://stackblitz.com/edit/effector-vite-react-template-dajnx8?file=tsconfig.json,src%2Fmodel.ts%3AL20,src%2Fanother-model.ts,src%2Fapplication.tsx

The culprit of the problem is `@typescript-eslint/parser` v6+
The change that make issue to appear in my local monorepo setup:
```ts
"@typescript-eslint/parser": "^5.59.0",
---->
"@typescript-eslint/parser": "^6.7.3",

```

I think this is related with [this one](https://github.com/effector/eslint-plugin/issues/144
)

**Comment by igorkamyshev** [comment]
It is about #144

Will be fixed 🫡

Let me close this one as duplicate
